packages/core/src/utils/basic.ts

```diff
@@ -1,7 +1,13 @@
+import type { Arrayify } from '../types'
+
 export function toArray<T>(value: T | T[] = []): T[] {
   return Array.isArray(value) ? value : [value]
 }
 
+export function toArrayReadonly<T>(value: T) {
+  return (Array.isArray(value) ? value : [value]) as Arrayify<T>
+}
```

Comment:
I am not sure if to have a runtime utility that only does type assessment (increase bundle size but not helpful at runtime). Also I think the name `Arrayify` is a bit confusing as it does not indicate it's readonly part. Maybe a `Readonly<ToArray<T>>` would be more straightforward.

Reply:
Reverted them so I could merge this feature first. We could discuss about it somewhere else

Reply:
tks ~